### PR TITLE
feat: add shared types

### PR DESCRIPTION
**Pull request by dmitryHarkusha93**

### CODE REVIEW

File: packages/shared-types/index.ts [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
   'sepolia' = '0xaa36a7',
   'mainnet' = '0x1',
   'goerli' = '0x5',
 }
 
+export enum ChainIdToNameEnum {
+  '0xaa36a7' = 'sepolia',
+  '0x1' = 'mainnet',
+  '0x5' = 'goerli',
+}
```
Comment (NasCorp) [inline packages/shared-types/index.ts:11]: @yoyq do you know any better way to describe this enums?

File: packages/shared-types/index.ts [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
 type BaseMonitor = {
-  name: string;
-  network: ChainEnum;
-  amount?: number;
+  name?: string;
+  network: keyof typeof ChainIdToNameEnum;
   intervalHours: string;
-  intervalMs?: number;
-  lastTransaction?: number;
+  intervalMs: number;
```
Comment (NasCorp) [inline packages/shared-types/index.ts]: @yoyq as I can see it was optional.

File: packages/shared-types/index.ts [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
+  contractAddress: string;
+  amount?: number;
```
Comment (NasCorp) [inline packages/shared-types/index.ts:20]: @yoyq are you sure about lastTransaction?

File: packages/shared-types/index.ts (hunk 1 above)
Comment (dmitryHarkusha93) [reply packages/shared-types/index.ts:11]: @NasCorp I haven't found any good solution

**Review by NasCorp — APPROVED** [review]

**Pull request merged**